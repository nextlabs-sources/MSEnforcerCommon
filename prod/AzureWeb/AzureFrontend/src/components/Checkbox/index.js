import React from 'react'
import propTypes from 'prop-types'

const Checkbox = ({ checked = false, handler, style }) => {
    const innerStyle = {
        display: 'inline-block',
        verticalAlign: 'top',
        border: checked ? '2px solid #06c' : '2px solid #999',
        borderRadius: '3px',
        boxSizing: 'border-box',
        backgroundColor: checked ? '#06c' : '#fff',
        backgroundClip: 'content-box',
        cursor: 'pointer',
    }

    const mergeStyle = {
        ...innerStyle,
        ...style,
    }

    const clickHandler = (e) => {
        e.stopPropagation()

        if (handler) {
            handler(!checked)
        }
    }
    
    return (
        <div style={mergeStyle} onClick={clickHandler} role='checkbox' aria-checked={checked} tabIndex={-1}>
            <svg focusable="false" width="100%" height="100%" viewBox="0 0 24 24" aria-hidden="true" preserveAspectRatio='xMinYMin meet'>
                <path fill='#fff' d="M9 16.17L4.83 12l-1.42 1.41L9 19 21 7l-1.41-1.41z"/>
            </svg>
        </div>
    )
}

export default Checkbox

Checkbox.propTypes = {
    checked: propTypes.bool,
    handler: propTypes.func.isRequired,
    style: propTypes.objectOf(propTypes.string),
}
