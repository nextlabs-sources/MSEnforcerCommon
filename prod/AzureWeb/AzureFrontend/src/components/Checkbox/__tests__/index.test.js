import React from 'react'
import renderer from 'react-test-renderer'
import Checkbox from '../index'

describe('Checkbox', () => {
    it('should render checked one', () => {
        const checkHandler = jest.fn()
        const tree = renderer.create(<Checkbox checked={true} handler={checkHandler}/>)
        const instance = tree.root

        instance.props.handler(true)

        expect(tree).toMatchSnapshot()
        expect(checkHandler.mock.calls.length).toBe(1)
        expect(checkHandler.mock.calls[0][0]).toBe(true)
    })

    it('should render unchecked one', () => {
        const checkHandler = jest.fn()
        const tree = renderer.create(<Checkbox checked={false} handler={checkHandler}/>)
        const instance = tree.root

        instance.props.handler(false)

        expect(tree).toMatchSnapshot()
        expect(checkHandler.mock.calls.length).toBe(1)
        expect(checkHandler.mock.calls[0][0]).toBe(false)
    })
})