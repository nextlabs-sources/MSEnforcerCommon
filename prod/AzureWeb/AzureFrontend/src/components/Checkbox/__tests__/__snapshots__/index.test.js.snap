// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox should render checked one 1`] = `
<div
  aria-checked={true}
  onClick={[Function]}
  role="checkbox"
  style={
    Object {
      "backgroundClip": "content-box",
      "backgroundColor": "#06c",
      "border": "2px solid #06c",
      "borderRadius": "3px",
      "boxSizing": "border-box",
      "cursor": "pointer",
      "display": "inline-block",
      "verticalAlign": "top",
    }
  }
  tabIndex={-1}
>
  <svg
    aria-hidden="true"
    focusable="false"
    height="100%"
    preserveAspectRatio="xMinYMin meet"
    viewBox="0 0 24 24"
    width="100%"
  >
    <path
      d="M9 16.17L4.83 12l-1.42 1.41L9 19 21 7l-1.41-1.41z"
      fill="#fff"
    />
  </svg>
</div>
`;

exports[`Checkbox should render unchecked one 1`] = `
<div
  aria-checked={false}
  onClick={[Function]}
  role="checkbox"
  style={
    Object {
      "backgroundClip": "content-box",
      "backgroundColor": "#fff",
      "border": "2px solid #999",
      "borderRadius": "3px",
      "boxSizing": "border-box",
      "cursor": "pointer",
      "display": "inline-block",
      "verticalAlign": "top",
    }
  }
  tabIndex={-1}
>
  <svg
    aria-hidden="true"
    focusable="false"
    height="100%"
    preserveAspectRatio="xMinYMin meet"
    viewBox="0 0 24 24"
    width="100%"
  >
    <path
      d="M9 16.17L4.83 12l-1.42 1.41L9 19 21 7l-1.41-1.41z"
      fill="#fff"
    />
  </svg>
</div>
`;
